jackson-jq/src/test/java/net/thisptr/jackson/jq/JsonQueryTest.java

```diff
@@ -196,6 +197,20 @@ private void test(final TestCase tc, final Version version) throws Throwable {
 			assertThat(failed).describedAs("unexpectedly succeeded").isTrue();
 	}
 
+	private void assertEquals(List<JsonNode> expected, List<JsonNode> actual, String description, Object... args) {
+		assertThat(actual).as(description, args)
+				.usingComparator((o1, o2) -> {
+					try {
+						String o1Json = JSON_MAPPER.writeValueAsString(o1);
+						String o2Json = JSON_MAPPER.writeValueAsString(o2);
+						return o1Json.compareTo(o2Json);
```

Comment:
It'd be better if you can compare the two values without converting them to strings first. We probably have to implement a new equality function that i) ignores the number type differences (`IntNode`, `LongNode`, `DoubleNode`, ...so that we don't have to distinguish between `3` and `3.0` in test cases) and ii) is sensitive to the order of the keys.

Reply:
Done, keep using `JsonNodeComparator`

Reply:
Sorry for the delay.

Unfortunately, we can't change how `JsonNodeComparator` compares two objects. The primary use of the comparator is `sort_by` and the comparison is explicitly specified as:
> first they're compared by comparing their sets of keys (as arrays in sorted order), and if their keys are equal then the values are compared key by key. -- https://stedolan.github.io/jq/manual/v1.6/#sort,sort_by(path_expression)

So we need to leave the comparator as it was and implement a new independent equality function solely for the tests. Let me handle the test part and I'll shortly merge this PR with my commit added.

Reply:
Can you undo the changes to `jackson-jq/src/main/java/net/thisptr/jackson/jq/internal/misc/JsonNodeComparator.java` and `jackson-jq/src/test/java/net/thisptr/jackson/jq/JsonQueryTest.java`, squash and force-push to the branch? I can edit the commit myself, but if I do that, this PR won't be marked as Merged after merge.

Reply:
Done.

Reply:
Thanks!